### PR TITLE
Fix superpmi collect failure for crossgen2 on linux x64 and windows x86

### PR DESCRIPTION
**Pull request by JulieLeeMSFT**
[spmi collect pipeline](https://dev.azure.com/dnceng/internal/_build?definitionId=977&_a=summary) started to fail since Feb'24 for crossgen2 on linux x64. Fix by including HelixRuntimeRid and download Dotnet Cli.

Edit: libraries crossgen2 was failing on windows x86 as well on partition 0 where it contained only ilc.exe. Fixed by ignoring ilc.exe because it does not have any assemblies to crossgen to.

cc @jkoritzinsky. PTAL.

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
Can you share a test run of the superpmi-collect pipeline with these changes (and a different JIT-EE guid to avoid overwriting existing collections)?

**Comment by JulieLeeMSFT** [comment]
Both tests are passing now:
- SuperPMI collect libraries crossgen2 linux x64 checked: [Log](https://dev.azure.com/dnceng/internal/_build/results?buildId=2493817&view=logs&j=db27b107-f5f1-52c4-1e78-5fafa21efcf9)
- SuperPMI collect libraries crossgen2 windows x86 checked [Log](https://dev.azure.com/dnceng/internal/_build/results?buildId=2493817&view=logs&j=3a8a826f-53c6-5325-2155-0051268bd6c2)

File: src/coreclr/scripts/superpmi_collect_setup.py [hunk 1]
```diff
@@ -183,6 +183,7 @@
     "vcruntime140.dll",
     "vcruntime140_1.dll",
     "xunit.console.exe",
+    "ilc.exe"
```
Comment (jakobbotsch) [inline src/coreclr/scripts/superpmi_collect_setup.py:186]: Nit: the list looks sorted, so can you place this after `ilasm`?

Comment (JulieLeeMSFT) [reply src/coreclr/scripts/superpmi_collect_setup.py]: Good catch. Done.

**Comment by JulieLeeMSFT** [comment]
@dotnet/jit-contrib, @jkoritzinsky, @jakobbotsch, it is ready to review.

**Review by jkoritzinsky — APPROVED** [review]

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**